src/test/java/teammates/test/cases/browsertests/InstructorFeedbackResponseCommentResultsPageUiTest.java

```diff
@@ -0,0 +1,194 @@
+
+        resultsPage.loadResultSectionPanel(1, 2);
+        resultsPage.addFeedbackResponseComment("showResponseCommentAddForm-1-1-1-1", "test comment 3");
+        resultsPage.verifyCommentRowContent("-1-1-1-1-1", "test comment 3", "Teammates Test");
+
+        ______TS("Typical case: edit existing feedback response comment");
+
+        resultsPage.editFeedbackResponseComment("-1-1-1-1-1", "edited test comment");
+        resultsPage.verifyCommentRowContent("-1-1-1-1-1", "edited test comment", "Teammates Test");
+        resultsPage.verifyHtmlMainContent("/instructorFeedbackResultsEditComment.html");
+
+        ______TS("Typical case: edit comment created by different instructor");
+
+        resultsPage.editFeedbackResponseComment("-1-0-1-1-1", "Comment edited by different instructor");
+        resultsPage.verifyCommentRowContent("-1-0-1-1-1", "Comment edited by different instructor", "Teammates Test");
+        resultsPage.verifyHtmlMainContent("/instructorFeedbackResultsEditCommentByDifferentInstructor.html");
+
+        ______TS("Typical case: delete existing feedback response comment");
+
+        resultsPage.deleteFeedbackResponseCommentInline("-1-1-1-1-1");
+        resultsPage.verifyRowMissing("-1-1-1-1-1");
+
+        resultsPage = loginToInstructorFeedbackResultsPage("CFResultsUiT.instr", "Open Session");
+        resultsPage.displayByRecipientGiverQuestion();
+        resultsPage.loadResultSectionPanel(0, 1);
+        resultsPage.verifyCommentRowContent("-0-0-1-1-2", "test comment 2", "Teammates Test");
+
+        ______TS("Typical case: add edit and delete successively");
+
+        resultsPage.displayByRecipientGiverQuestion();
+        resultsPage.loadResultSectionPanel(0, 1);
+        resultsPage.addFeedbackResponseComment("showResponseCommentAddForm-0-0-1-1", "successive action comment");
+        resultsPage.verifyCommentRowContent("-0-1-0-1-3", "successive action comment", "Teammates Test");
+
+        resultsPage.editFeedbackResponseComment("-0-1-0-1-3", "edited successive action comment");
+        resultsPage.verifyCommentRowContent("-0-1-0-1-3", "edited successive action comment",
+                "Teammates Test");
+        resultsPage.clickVisibilityOptionForResponseCommentAndSave("responseCommentRow-0-1-0-1-3", 1);
+
+        resultsPage.deleteFeedbackResponseCommentInline("-0-1-0-1-3");
+        resultsPage.verifyRowMissing("-0-0-1-1-3");
+
+        resultsPage = loginToInstructorFeedbackResultsPage("CFResultsUiT.instr", "Open Session");
+        resultsPage.displayByRecipientGiverQuestion();
+        resultsPage.loadResultSectionPanel(0, 1);
+        resultsPage.verifyCommentRowContent("-0-0-1-1-2", "test comment 2", "Teammates Test");
+        resultsPage.verifyRowMissing("-0-0-1-1-3");
+        resultsPage.verifyHtmlMainContent("/instructorFeedbackResultsDeleteComment.html");
+    }
+
+    private InstructorFeedbackResultsPage loginToInstructorFeedbackResultsPage(String instructorName, String fsName) {
```

Comment:
I think it will add additional coupling as methods like `createUrl()` and `loginToAdminPage()` are use from the parent class `BaseUiTestCase.java`. 

Also, it will require more parameters like `testData` to be passed increasing the coupling

Reply:
Good point, I missed that.

Although there's some duplication it seems like we can't do much within this PR. Seems like there's a problem where the UI tests have an unnecessary coupling with things that belong more to pages.